### PR TITLE
Return BadRequestError from beginning

### PR DESCRIPTION
**Pull request by vancexu**
- fix improper error return type in history
- return bad request error from beginning
- update tests to use valid runID
- add unit tests for invalid runID and empty runID

### CODE REVIEW

**Comment by vancexu** [comment]
related https://github.com/uber/cadence/issues/509

File: service/history/handler.go [hunk 1]
```diff
@@ -845,3 +875,13 @@ func createShardOwnershipLostError(currentHost, ownerHost string) *hist.ShardOwn
 
 	return shardLostErr
 }
+
+func validateToken(token *common.TaskToken) error {
```
Comment (wxing1292) [inline service/history/handler.go]: validate task token

File: service/history/historyEngine2_test.go [hunk 2]
```diff
@@ -190,7 +190,7 @@ func (s *engine2Suite) TestRecordDecisionTaskStartedIfNoExecution() {
 func (s *engine2Suite) TestRecordDecisionTaskStartedIfGetExecutionFailed() {
 	workflowExecution := &workflow.WorkflowExecution{
 		WorkflowId: common.StringPtr("wId"),
-		RunId:      common.StringPtr("rId"),
+		RunId:      common.StringPtr(uuid.New()),
```
Comment (wxing1292) [inline service/history/historyEngine2_test.go]: validRunID?
